cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java

```diff
@@ -169,7 +169,14 @@ public void initialize(MapReduceContext context) throws Exception {
     BatchSourceContext sourceContext = new MapReduceSourceContext(context, mrMetrics,
                                                                   new DatasetContextLookupProvider(context),
                                                                   sourceName, context.getRuntimeArguments());
+
+    ClassLoader oldClassLoader = Thread.currentThread().getContextClassLoader();
+    Thread.currentThread().setContextClassLoader(batchSource.getClass().getClassLoader());
+
     batchSource.prepareRun(sourceContext);
+
+    Thread.currentThread().setContextClassLoader(oldClassLoader);
```

Comment:
you need to make sure the context classloader gets set back when exceptions getting thrown in the plugin methods. So something like:

```
try {
  batchSource.prepareRun(sourceContext);
} finally {
  Thread.currentThread().setContextClassLoader(oldClassLoader);
}
```